lib/src/data/tutor_availability.dart

```diff
@@ -0,0 +1,55 @@
+import "utils.dart";
+
+extension type TutorID(String value) {}
+extension type CourseID(String value) {}
+
+class TutorAvailability {
+  final TutorID tutorId;
+  final String department;
+  final List<CourseID> availableCourses;
```

Comment:
```suggestion
  final Set<CourseID> availableCourses;
```

Sets are better for cases like this because they have O(1) searches and we don't _need_ an order. 

We should also think about how we can efficiently query for courses that have a tutor. CourseID is fine for now but we might need to change this later (because RegisterBlast doesn't use our IDs)